cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLGenerator.scala

```diff
@@ -0,0 +1,654 @@
+       |);
+      """.stripMargin
+  }
+
+  protected def fieldType2ADMType(field: Field): String = {
+    field.dataType match {
+      case DataType.Number => "double"
+      case DataType.Time => "datetime"
+      case DataType.Point => "point"
+      case DataType.Boolean => "tinyint"
+      case DataType.String => "varchar(255)"
+      case DataType.Text => "text"
+      case DataType.Json => "json"
+      case DataType.Bag => s"{{${fieldType2ADMType(Field("", field.asInstanceOf[BagField].innerType))}}}"
+      case DataType.Hierarchy => ???
+      case DataType.Record => ???
+    }
+  }
+
+  def parseAppend(append: AppendView, schemaMap: Map[String, AbstractSchema]): String = {
+    s"""
+       |replace into $quote${append.dataset}$quote (
+       |${parseQuery(append.query, schemaMap)}
+       |)""".stripMargin.replace("T", " ").replace("Z", "")
+  }
+
+  def parseUpsert(q: UpsertRecord): String = {  //TODO: generalization
+    val x = q.records.value
+    var queryResult = ArrayBuffer.empty[String]
+    x.foreach {
+      record =>
+        val name: String = (record \ "name").as[JsString].value
+        val schema: JsValue = (record \ "schema").as[JsValue]
+        val dataInterval: JsValue = (record \ "dataInterval").as[JsValue]
+        val stats: JsValue = (record \ "stats").as[JsValue]
+        queryResult += (s"('${name}','${schema}','${dataInterval}','${stats}')")
+    }
+    s"""
+       |replace into $quote${q.dataset}$quote (`name`, `schema`, `dataInterval`, `stats`) values
+       |${queryResult.mkString(",")}
+       |""".stripMargin
+  }
+
+  protected def parseDelete(delete: DeleteRecord, schemaMap: Map[String, AbstractSchema]): String = {
+    if (delete.filters.isEmpty) {
+      throw new QueryParsingException("Filter condition is required for DeleteRecord query.")
+    }
+    val exprMap: Map[String, FieldExpr] = initExprMap(delete.dataset, schemaMap)
+    val queryBuilder = new StringBuilder()
+    queryBuilder.append(s"delete from $quote${delete.dataset}$quote $sourceVar")
+    parseFilter(delete.filters, exprMap, Seq.empty, queryBuilder)
+    queryBuilder.toString()
+  }
+
+  protected def parseDrop(query: DropView, schemaMap: Map[String, AbstractSchema]): String = {
+    s"drop table if exists ${query.dataset}"
+  }
+
+  def calcResultSchema(query: Query, schema: Schema): Schema = {
+    if (query.lookup.isEmpty && query.groups.isEmpty && query.select.isEmpty) {
+      schema.copy()
+    } else {
+      ???
+    }
+  }
+
+  protected def initExprMap(dataset: String, schemaMap: Map[String, AbstractSchema]): Map[String, FieldExpr] = {
+    val schema = schemaMap(dataset)
+    schema.fieldMap.mapValues { f =>
+      f.dataType match {
+        case DataType.Json if f.name.contains(".") =>
+            FieldExpr(s"$sourceVar.$quote${f.name}$quote", s"$sourceVar.${f.name.split("\\.", 2).head + "->\"$." + f.name.split("\\.", 2)(1) + '"'}")
```

Comment:
Change the defExpr when calling Json's subfield. 
e.g.: Convert user.id to user->"$.id"
I'm considering adding a function only for Json's field's transform, so this line can stop repeating.